### PR TITLE
Dialog codemod

### PR DESCRIPTION
**Pull request by AntanasBuk**
## Changes

Codemod to accommodate changes required for `Dialog` deprecation and moving to iTwinUI `Dialog`. This does not include codemods for similar `Dialog` JSXElements like `ModelessDialog` or `GlobalDialog`.
Also includes structural changes to codemod tool as well as extended functionality.
 
## Testing

Extensive tests to confirm codemod behavior

### CODE REVIEW

File: ui/core-react/src/core-react/dialog/Dialog.tsx [hunk 1]
```diff
@@ -391,3 +391,76 @@ export class Dialog extends React.Component<DialogProps, DialogState> {
     }
   };
 }
+
+/**
+ * A temporary function for converting DialogButtonDef items to Button elements
+ * @public
```
Comment (raplemie) [inline ui/core-react/src/core-react/dialog/Dialog.tsx:397]: Not sure we want to make this public

File: common/config/rush/pnpm-lock.yaml
Comment (raplemie) [inline common/config/rush/pnpm-lock.yaml:1]: We'll need to revert the changes on this file

File: tools/codemod/src/utils/Extensions.ts [hunk 2]
```diff
@@ -2,33 +2,134 @@
...
   toExpressionName,
   useCallExpression,
 } from "./CallExpression";
-import { useImportDeclaration } from "./ImportDeclaration";
 import {
-  findRootIdentifiers,
-  sortSpecifiers,
-  useImportSpecifier,
-} from "./ImportSpecifier";
+  ImportDeclarationCollection,
```
Comment (raplemie) [inline tools/codemod/src/utils/Extensions.ts:14]: Unused

File: tools/codemod/src/utils/Extensions.ts [hunk 3]
```diff
@@ -2,33 +2,134 @@
  * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
  * See LICENSE.md in the project root for license terms and full copyright notice.
  *--------------------------------------------------------------------------------------------*/
-import type { ASTNode, Collection, JSCodeshift } from "jscodeshift";
+import type { ASTNode, ASTPath, Collection, JSCodeshift } from "jscodeshift";
+import type { ASTNode as ASTNodeType, Type } from "ast-types/lib/types";
+import type { NodePath } from "ast-types/lib/node-path";
```
Comment (raplemie) [inline tools/codemod/src/utils/Extensions.ts:7]: Implementation commented out, not used.

File: tools/codemod/src/utils/Extensions.ts [hunk 4]
```diff
@@ -2,33 +2,134 @@
...
+              const childNodePathName =
+                typeof childPath.name === "number"
+                  ? childPath.parentPath.name
+                  : childPath.name;
+              return nodePathName === childNodePathName;
+            }
+            return true;
+          })
+          .paths()
+      );*/
```
Comment (raplemie) [inline tools/codemod/src/utils/Extensions.ts:113]: Is it better ? Is it a reference ? Should this be used or the comment removed ?

Comment (AntanasBuk) [reply tools/codemod/src/utils/Extensions.ts:113]: The commented out code is theoretically slower since it finds all children of children of children... and then filters only the first child layer. The code above is a tweaked implementation of jscodeshift find method so it would stop after finding only the first layer of children (immediate children). So it is up to you to decide which one to keep (above one is faster, commented one is more readable)

**Comment by GerardasB** [comment]
Will not be merged until we work on Dialog deprecation. Changes are available in a branch https://github.com/iTwin/appui/tree/dialog-codemod
